kensu/requests/api.py

```diff
@@ -11,33 +13,49 @@ def wrapper(*args, **kwargs):
 
         kensu = KensuProvider().instance()
 
-        result_pk = DataSourcePK(location=result.url,
-                                 physical_location_ref=kensu.default_physical_location_ref)
-        result_ds = DataSource(name=result.url, format='API',
-                               pk=result_pk)._report()
+        try:
+            fields_dict = flatten(result.json())
 
+            # Construct the file location
 
-        fields_dict = flatten(result.json())
+            url = kwargs['url'] if 'url' in kwargs else args[0]
+            try:
+                params = kwargs['params'] if 'params' in kwargs else args[1]
+            except:
+                params = None
 
-        fields = [FieldDef(name = k,  field_type=fields_dict[k], nullable=True ) for k in  fields_dict]
-        sc_pk = SchemaPK(result_ds.to_ref(),
-                         fields=fields)
+            if params is not None:
+                location = url + '?' + ('&').join([e+'={}' for e in params.keys()])
+            else:
+                location = url
 
-        result_sc = Schema(name="schema:" + result_ds.name, pk=sc_pk)._report()
 
-        result.__class__ = Response
-        result.ksu_schema = result_sc
-        result.ds_location = result.url
-        try:
-            import json
-            d = json.loads(result.text)
-            count_json = len(d)
-            stats = {'count':count_json}
+            result_pk = DataSourcePK(location=location,
+                                     physical_location_ref=kensu.default_physical_location_ref)
+            result_ds = DataSource(name=location, format='API',
+                                   pk=result_pk)._report()
+
+            fields = [FieldDef(name = k,  field_type=fields_dict[k], nullable=True ) for k in  fields_dict]
+            sc_pk = SchemaPK(result_ds.to_ref(),
+                             fields=fields)
+
+            result_sc = Schema(name="schema:" + result_ds.name, pk=sc_pk)._report()
+
+            result.__class__ = Response
```

Comment:
hmm, we do not modify the `result` class (just add monkey attributes), so this line is very likely not needed